backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala

```diff
@@ -481,10 +481,10 @@ object BackendSettings extends BackendSettingsApi {
   private def checkMaxBatchSize(nativeConf: java.util.Map[String, String]): Unit = {
     if (nativeConf.containsKey(GlutenConfig.GLUTEN_MAX_BATCH_SIZE_KEY)) {
       val maxBatchSize = nativeConf.get(GlutenConfig.GLUTEN_MAX_BATCH_SIZE_KEY).toInt
-      if (maxBatchSize > MAXIMUM_BATCH_SIZE) {
+      if (maxBatchSize >= MAXIMUM_BATCH_SIZE) {
```

Comment:
The check in native shuffle writer requires the split buffer (aka. output batch size from shuffle writer) <= 32k, which means the maximum batch size in the pipeline can equal to 32k. But this change makes the maximum batch size < 32k.

Reply:
Got it

Reply:
@marin-ma we should remove the check here, right?